src/proxify.js

```diff
@@ -0,0 +1,81 @@
+
+const getOptions = options => ({
+  name: 'object',
+  seal: true,
+  sealGet: false,
+  deprecated: [],
+  exposeApi: false,
+  apiNamespace: '__proxy__',
+  ...options
+});
+
+const getPropertyNames = (obj) => {
+  if (!obj) {
+    return [];
+  }
+  const methods = Object.getOwnPropertyNames(obj.prototype || obj);
+  const proto = Object.getPrototypeOf(obj);
+  return deleteDuplicates([
+    ...methods,
+    ...getPropertyNames(proto)
+  ]);
+};
+
+const deleteDuplicates = arr => [...new Set(arr)];
+
+const proxify = (obj, opts = {}) => {
+  if (!obj || (typeof obj !== 'object' && typeof obj !== 'function')) {
+    throw Error('proxify only applies on non-null object');
+  }
+
+  const options = getOptions(opts);
+  const properties = ['inspect'];
+  
+  if (options.exposeApi) {
+    obj[options.apiNamespace] = {
+      registerProps: name => {
+        if (!properties.includes(name)) {
+          properties.push(name);
+        }
+      },
+      unregisterProps: name => {
+        const index = properties.indexOf(name);
+        if (index !== -1) {
+          properties.splice(index, 1);
+        }
+      },
+      hasProp: name => properties.includes(name)
+    };
+  }
+  
+  properties.push.apply(properties, deleteDuplicates([
+    ...Object.getOwnPropertyNames(obj),
+    ...getPropertyNames(obj)
+  ]));
+
+  const handler = {
+    get: (target, name) => {
+      if (options.sealGet && typeof name === 'string' && !properties.includes(name)) {
```

Comment:
you should remove the typecheck made on `name`: 
* it can only be a string or a symbol anyway, since this is a getter trap
* if I pass a non-existing symbol to your handler, I won't get any error even with `sealGet` active, which does not seem to be the intended behavior